File: dev/SPANetTest/TOne.cs


using System;
using SocketProAdapter;
using SocketProAdapter.ClientSide;

public class CTOne : CAsyncServiceHandler
{
    public CTOne()
        : base(TOneConst.sidCTOne)
    {
    }

    public int QueryCount()
    {
        int QueryCountRtn;
        bool bProcessRy = ProcessR1(TOneConst.idQueryCountCTOne, out QueryCountRtn);
        return QueryCountRtn;
    }

    public int QueryGlobalCount()
    {
        int QueryGlobalCountRtn;
        bool bProcessRy = ProcessR1(TOneConst.idQueryGlobalCountCTOne, out QueryGlobalCountRtn);
        return QueryGlobalCountRtn;
    }

    public int QueryGlobalFastCount()
    {
        int QueryGlobalFastCountRtn;
        bool bProcessRy = ProcessR1(TOneConst.idQueryGlobalFastCountCTOne, out QueryGlobalFastCountRtn);
        return QueryGlobalFastCountRtn;
    }

    public void Sleep(int nTime)
    {
        bool bProcessRy = ProcessR0(TOneConst.idSleepCTOne, nTime);
    }

    public object Echo(object objInput)
    {
        object EchoRtn;
        bool bProcessRy = ProcessR1(TOneConst.idEchoCTOne, objInput, out EchoRtn);
        return EchoRtn;
    }

    public bool EchoEx(sbyte[] str, string wstr, MyStruct ms, out sbyte[] strOut, out string wstrOut)
    {
        bool EchoExRtn;
        bool bProcessRy = ProcessR3(TOneConst.idEchoExCTOne, str, wstr, ms, out strOut, out wstrOut, out EchoExRtn);
        return EchoExRtn;
    }
}

File: dev/vbtest/TOne.vb
Imports System
Imports SocketProAdapter
Imports SocketProAdapter.ClientSide

public class CTOne : Inherits CAsyncServiceHandler
	Public Sub New()
		MyBase.New(TOneConst.sidCTOne)
	End Sub

	Public Function QueryCount() As Integer
		Dim QueryCountRtn As Integer
		Dim bProcessRy as Boolean = ProcessR1(TOneConst.idQueryCountCTOne, QueryCountRtn)
		return QueryCountRtn
	End Function

	Public Function QueryGlobalCount() As Integer
		Dim QueryGlobalCountRtn As Integer
		Dim bProcessRy as Boolean = ProcessR1(TOneConst.idQueryGlobalCountCTOne, QueryGlobalCountRtn)
		return QueryGlobalCountRtn
	End Function

	Public Function QueryGlobalFastCount() As Integer
		Dim QueryGlobalFastCountRtn As Integer
		Dim bProcessRy as Boolean = ProcessR1(TOneConst.idQueryGlobalFastCountCTOne, QueryGlobalFastCountRtn)
		return QueryGlobalFastCountRtn
	End Function

	Public Sub Sleep(ByVal nTime As Integer) 
		Dim bProcessRy as Boolean = ProcessR0(TOneConst.idSleepCTOne, nTime)
	End Sub

	Public Function Echo(ByVal objInput As Object) As Object
		Dim EchoRtn As Object = Nothing
		Dim bProcessRy as Boolean = ProcessR1(TOneConst.idEchoCTOne, objInput, EchoRtn)
		return EchoRtn
	End Function

    Public Function EchoEx(ByVal str As SByte(), ByVal wstr As String, ByRef strOut As SByte(), ByRef wstrOut As String) As Boolean
        'Please implement IUSerializer for the class MyStruct
        Dim EchoExRtn As Boolean
        Dim bProcessRy As Boolean = ProcessR3(TOneConst.idEchoExCTOne, str, wstr, strOut, wstrOut, EchoExRtn)
        Return EchoExRtn
    End Function
End Class
